omod/src/main/java/org/openmrs/module/pihmalawi/web/taglibs/EMastercardAccessTag.java

```diff
@@ -77,7 +78,11 @@ public int doStartTag() throws JspException {
 			}
 
             // Ensure no more than one initial encounter is found
-			List<Encounter> initials = Context.getEncounterService().getEncounters(p, null, null, null, null, Arrays.asList(initialEncounterType), null, false);
+			EncounterSearchCriteriaBuilder encounterSearchCriteriaBuilder = new EncounterSearchCriteriaBuilder();
+			encounterSearchCriteriaBuilder.setPatient(p);
+			encounterSearchCriteriaBuilder.setEncounterTypes(Arrays.asList(initialEncounterType));
+			encounterSearchCriteriaBuilder.setIncludeVoided(false);
+			List<Encounter> initials = Context.getEncounterService().getEncounters(encounterSearchCriteriaBuilder.createEncounterSearchCriteria());
 			if (initials.size() > 1) {
```

Comment:
Would be worth moving this to a utility method since you have duplicated this same logic with just different encounter types in several places here and below.

Reply:
Thanks @mseaton ! I have added a getEncounters method in the Utils class, and refactored this code.